README.md

```diff
@@ -74,81 +121,184 @@ always ready.
+
+Registering `zephyr-image` Update Module (compiled in by default):
+```c
+mender_zephyr_image_register_update_module());
 ```
-The mac address is an arbitrary identifier. You can use anything as long as it is unique for each device.
 
-The tenant token can be found under `My organization` in hosted Mender, where it's called `Organization token`.
+#### Enabling Inventory
+Inventory is enabled/disabled by setting the `MENDER_CLIENT_INVENTORY` option.
 
-### Creating an Artifact
-Create an artifact (remember to disable compression):
+To use inventory, you will need a `mender_keystore_t` struct, which is defined in [mender-client.h](include/mender-client.h).
+This struct contains the inventory of the client, and is enabled in the code by calling `mender_inventory_set`.
+
+Example:
+```c
+mender_keystore_t inventory[] = { { .name = "demo", .value = "demo" }, { .name = "foo", .value = "bar" }, { .name = NULL, .value = NULL } };
+mender_inventory_set(inventory));
 ```
-./mender-artifact write module-image -T zephyr-image --compression none --artifact-name <artifact_name> --device-type <device_type> --file <file_name>
+
+#### Activating the Client
+Finally, you can activate the client by calling `mender_client_activate`:
+```c
+mender_client_activate();
 ```
-The `device_type` in the artifact has to match the `device_type` used when running the client.
+After the client is activated, it will run on a workqueue thread, either a dedicated one
+(default) or the system one (configurable with the options starting with `CONFIG_SYSTEM_WORKQUEUE_`).
+From this thread, the client will regularly poll for updates and submit inventory.
+
+### Building the client
+
+See the [building section](https://github.com/mendersoftware/mender-mcu-integration?tab=readme-ov-file#building-the-zephyr-project-mender-reference-app)
+in mender-mcu-integration for a list of the boards that we have tested, and for examples
+on how to build for them.
+
+### Creating a Mender Artifact
 
+NOTE: Compression of artifacts is **not** supported, which is why we use `--compression none`
+
+After building the client, you can create a [Mender Artifact](https://docs.mender.io/overview/artifact).
+
+The update module must correspond with one of the ones registered during initialization of the client.
+The device type must match the one set at build time - if you're unsure what this is, you can
+check the value of `MENDER_DEVICE_TYPE` in the menuconfig, by default it uses Zephyr's `BOARD`.
+
+The Artifact Name is arbitrary, but should be unique.
+
+```bash
+UPDATE_MODULE=<update_module>
+ARTIFACT_NAME=<artifact_name>
+DEVICE_TYPE=<device_type>
+
+mender-artifact write module-image \
+  --type $UPDATE_MODULE \
+  --file build/zephyr/zephyr.signed.bin \
+  --artifact-name $ARTIFACT_NAME \
+  --device-type $DEVICE_TYPE \
+  --compression none
+```
 ### Deployment
 After creating and uploading the artifact to the server, you should be able to deploy it to the device.
```

Comment:
Would make this more confident :) "should be" doesn't instill confidence

... upload the Artifact to the Mender server and deploy it to your device.